Assets/Scripts/Models/Time/TimeManager.cs

```diff
@@ -130,6 +134,16 @@ public void Update(float time)
         // Systems that update every frame.
         InvokeEvent(EveryFrame, time);
 
+        if (nextFrameActions.Count > 0)
+        {
+            for (int i = 0; i < nextFrameActions.Count; i++)
```

Comment:
I personally simply like the verbosity and more natural feel of a `foreach (String param in params)` more than the a little more clunky `for (String param : params)`. And yes, I do realise that Java is shorter there. 

But enough of that tangent here.